124-stretch-cluster.md

```diff
@@ -0,0 +1,1795 @@
+- **Automation:** Python scripts for automated test execution and result collection
+
+**Test Scenarios:**
+1. **Baseline:** same datacenter deployment (validates full throughput)
+2. **Latency Sensitivity:** Progressive latency injection (0ms, 10ms, 50ms, 100ms)
+3. **Breaking Point:** Binary search to find Kafka stability threshold
+
+**Conservative Test Methodology:**
+Latency sensitivity tests use Chaos Mesh NetworkChaos CR to inject artificial delay between **all pods in all clusters**, including intra-cluster communication. 
+This represents a **worst-case scenario** more pessimistic than real world deployments, where only cross cluster communication experiences added latency. 
+Real production performance in same datacenter multi-AZ deployments will be significantly better than these conservative test results.
+
+**Validation Criteria:**
+- same datacenter deployment maintains >= 90% of single cluster throughput
+- P99 latency remains < 500ms for same site deployments
+- Zero pod crashes or restarts under normal conditions
+- Graceful degradation under network latency (no crashes until 750ms threshold)
+
+**Validation Results:**
+
+Performance testing using OMB and Chaos Mesh validates the performance characteristics described in the "Performance Characteristics and Operational Guidelines" section. 
+The testing methodology can be documented and can be replicated by users to validate their own deployments.
+
+#### Validation Approach
+
+**Operator and Kafka Upgrades:**
+
+Standard Strimzi upgrade testing applies to stretch clusters. 
+The implementation includes validation that all operators run compatible versions, and standard rolling update mechanisms work across multiple clusters.
+
+**Plugin Testing:**
+
+Reference implementations (NodePort, LoadBalancer) maintained by Strimzi include comprehensive test coverage. 
+Third party plugins (e.g., MCS) are maintained independently with their own testing approaches.
+The SPI design allows plugin authors to validate their implementations without requiring changes to core Strimzi tests.
+
+### Additional Considerations
+
+#### External Access
+
+External listener types (Route, Ingress, LoadBalancer, NodePort) work in stretch mode:
+- Each cluster exposes its own external endpoints
+- Kafka CR status lists all bootstrap addresses from all clusters
+- Clients can connect to any cluster's bootstrap address
+- Kafka handles client redirection to appropriate brokers
+
+
+#### Rack Awareness
+
+Rack awareness works across clusters by using topology labels:
+- Each cluster can have different zone labels
+- The `strimzi.io/stretch-cluster-alias` annotation serves as the primary topology key
+- Additional zone labels within each cluster provide finer-grained replica distribution
+
+Example topology configuration:
+
+```yaml
+spec:
+  kafka:
+    rack:
+      topologyKey: topology.kubernetes.io/zone
+```
+
+Kafka distributes replicas considering both cluster boundaries and zone labels within clusters.
```

Comment:
I think you do not necessarily want to have different racks in different clusters just by definition. This should probably rely on something like described in https://github.com/strimzi/proposals/pull/184.